src/nnfusion/frontend/onnx_import/op/one_hot.hpp

```diff
@@ -42,16 +42,22 @@ namespace nnfusion
                     auto off_on_values = input_indexes[2];
 
                     std::vector<int64> depth_vec;
-                    NNFUSION_CHECK(GetValueFromNGraphOp(depth.gnode, &depth_vec));
+                    NNFUSION_CHECK(GetValueFromNGraphOp(depth.gnode, &depth_vec))
+                        << "The input of depth needs to be constant values.";
                     NNFUSION_CHECK(depth_vec.size() == 1);
 
-                    std::vector<int64> off_on_values_vec;
-                    NNFUSION_CHECK(GetValueFromNGraphOp(off_on_values.gnode, &off_on_values_vec));
+                    // we currently only handle numeric data types by converting them to double first,
+                    // then to real target type in codegen
+                    std::vector<double> off_on_values_vec;
```

Comment:
`int_64` is not representable by `double`, use `long double` if you don't want to lose data.